internal/pkg/gateway/registry/registry.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package registry
+
+import (
+	"crypto/x509"
+	"fmt"
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/common/flogging"
+	pp "github.com/hyperledger/fabric/core/peer"
+	"github.com/hyperledger/fabric/discovery"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// Config represents the startup configuration of the gateway
+type Config interface {
+}
+
+type registry struct {
+	localEndorser peer.EndorserClient
+	discovery     *discovery.Service
+}
+
+var logger = flogging.MustGetLogger("gateway.registry")
+
+func NewRegistry(discovery *discovery.Service) (*registry, error) {
```

Comment:
I don't get it. :) I commented just yesterday on how an exported constructor returning an unexported type makes little to no sense.